model/account/accounts.go

```diff
@@ -239,6 +248,33 @@ func PushAccountDeletedJob(jobsSystem job.JobSystem, db prefixer.Prefixer, accou
 	})
 }
 
+// ComputeName tries to use the value of the `auth` attribute pointed by the
+// value of the `identifier` attribute as the Account name and set it in the
+// JSON document.
+//
+// See https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.accounts.md#about-the-name-of-the-account
+func ComputeName(doc couchdb.JSONDoc) {
+	auth, ok := doc.M["auth"].(map[string]interface{})
+	if !ok || auth == nil {
+		doc.M["name"] = ""
+		return
+	}
+
+	identifier, ok := doc.M["identifier"].(string)
+	if !ok || identifier == "" {
+		if login, ok := auth["login"].(string); ok {
+			doc.M["name"] = login
+		} else {
+			doc.M["name"] = ""
+		}
+		return
+	}
+
+	if name, ok := auth[identifier].(string); ok {
+		doc.M["name"] = name
+	}
```

Comment:
It is a bit weird: you are forcing `doc.M["name"]` to a blank string when there are no auth or no identifier, but not here when the field pointed by the identifier is missing. Is it voluntary?

Reply:
I assumed the result of the type assertion would be an empty string in this case. But I might even not need to force this value at all?!
`""` is the default value of a string field, right?

Reply:
Right, the default value is `""`. You only need to assign to a blank string if there was a previous value that you want to remove.